src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -436,10 +436,10 @@ public static object MarkAsSecurityArtifact(object arg, Func<object, string> cal
         /// <param name="innerException"><see cref="Exception"/></param>
         /// <param name="message">The log message.</param>
         /// <param name="args">An object array that contains zero or more objects to format.</param>
-        private static LogEntry WriteEntry(EventLogLevel eventLogLevel, Exception innerException, string message, params object[] args)
+        private static LogEntry WriteEntry(EventLogLevel eventLogLevel, Exception? innerException, string message, params object[]? args)
         {
             if (string.IsNullOrEmpty(message))
-                return null;
+                return null!;
```

Comment:
The issue, I think, is this will return null, so adding a null return type `LogEntry?` means things like this 
```csharp
if (Logger.IsEnabled(eventLogLevel))
                Logger.Log(WriteEntry(eventLogLevel, exception.InnerException, exception.Message, null));
``` 
could have null references.

I'm wondering why we return null here anyway if message is empty? We could remove the null check on the message, looks like the only place where it's not passed is here, and we can check for null before:
`Logger.Log(WriteEntry(eventLogLevel, exception.InnerException, exception.Message ?? string.Empty, null));` and pass in an empty string instead. 

Not sure the right thing to do here, but it's a bit tricky.

Reply:
@jennyf19 that is a solution however, an empty string and a Null string are not equal so, I opted to keep the current flow by using !, not the best solution or a solution at all but, given the deadline I opted for keeping the current flow untouched.

Reply:
can we do `message = string.Empty;`?

Reply:
this one still needs to be figured out, we can't return null here.